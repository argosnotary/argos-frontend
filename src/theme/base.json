{
  "fontFamily": "'Fira Sans'",
  "fontFile": "/fonts/fira-sans-v10-latin-regular",
  "button": {
    "bgColor": "#1779ba",
    "textColor": "#fff"
  },
  "cancelButton": {
    "bgColor": "transparent",
    "textColor": "#8D99AE",
    "borderColor": "#8D99AE",
    "hover": {
      "bgColor": "#8D99AE",
      "textColor": "#fff",
      "borderColor": "transparent"
    }
  },
  "transparentButton": {
    "default": {
      "transparentButtonBgColor": "#fff",
      "transparentButtonTextColor": "#1779ba",
      "transparentButtonBorderColor": "#1779ba"
    },
    "hover": {
      "transparentButtonBgColor": "#1779ba",
      "transparentButtonTextColor": "#fff",
      "transparentButtonBorderColor": "#1779ba"
    }
  },
  "alternateLoader": {
    "color": "#808080"
  },
  "loaderIcon": {
    "color": "#1779ba"
  },
  "loaderButton": {
    "loadingColor": "#fff"
  },
  "dataCheckbox": {
    "loaderColor": "#1779ba"
  },
  "searchInput": {
    "searchResultsBgColor": "#fff",
    "searchResultEntryHoverBg": "#e8feff",
    "selectedEntryBorderColor": "#e0e0e0",
    "searchResultsBorderColor": "#e0e0e0",
    "loaderIconColor": "#1779ba"
  },
  "collapsibleContainer": {
    "iconColor": "gray",
    "headerBgColor": "#e0e0e0",
    "bodyBorderColor": "#e0e0e0"
  },
  "selectList": {
    "defaultItemBgColor": "#fff",
    "selectedItemBgColor": "#e8feff",
    "iconColor": "#8d99ae"
  },
  "panel": {
    "icons": {
      "shrinkIcon": {
        "color": "#1779ba"
      },
      "enlargeIcon": {
        "color": "#1779ba"
      }
    }
  },
  "modal": {
    "bgColor": "#fff",
    "button": {
      "default": {
        "color": "#1779ba"
      },
      "hover": {
        "color": "#1779ba"
      }
    }
  },
  "passwordView": {
    "clipboardIcon": {
      "color": "#8D99AE",
      "hoverBgColor": "#e8feff"
    }
  },
  "copyInput": {
    "bgColor": "#e0e0e0"
  },
  "notificationsList": {
    "notificationBgColor": "#fff",
    "iconSize": 22,
    "iconColor": "#fff",
    "colors": {
      "WARNING": "orange",
      "ERROR": "red",
      "SUCCESS": "green"
    }
  },
  "layoutBuilder": {
    "segmentContainerBorderColor": "#1779ba",
    "segmentContainerTitleBgColor": "#1779ba",
    "iconColor": "#1779ba",
    "stepContainerBgColor": "#e0e0e0",
    "stepContainerTitleBorderColor": "#1779ba",
    "addStepButtonBorderColor": "#1779ba",
    "segmentTitleBgColor": "#fff",
    "segmentTitleHoverBorderColor": "#1779ba",
    "stepTitleBgColor": "#fff",
    "stepTitleHoverBorderColor": "#1779ba",
    "segmentsContainerTitleColor": "#fff"
  },
  "treeEditor": {
    "textColor": "#2e2e2e",
    "bgColor": "#f1f1f1",
    "borderColor": "#e0e0e0",
    "loaders": {
      "onPageLoad": {
        "color": "#1779ba"
      },
      "onFetchChildren": {
        "color": "#1779ba"
      }
    },
    "iconColors": {
      "addRootNode": "#1779ba",
      "expandNode": "#1779ba",
      "label": "#8D99AE",
      "chain": "#8D99AE",
      "robot": "#8D99AE"
    },
    "treeHeadLabel": {
      "bgColor": "#e8feff"
    },
    "nodeContextMenuContainer": {
      "bgColor": "fff",
      "borderColor": "#e5e5e5"
    },
    "nodeContextMenuItem": {
      "iconColor": "#8D99AE",
      "bgColor": "#fff",
      "hover": {
        "bgColor": "#e8feff"
      }
    },
    "breadCrumb": {
      "textColor": "#1779ba"
    },
    "lastBreadCrumb": {
      "textColor": "#000"
    },
    "nodeContextMenuItemSeparator": {
      "borderColor": "#e5e5e5"
    },
    "nodeContextMenuIconSeparator": {
      "borderColor": "#8D99AE"
    }
  },
  "dashboardNavBar": {
    "link": {
      "textColor": "#2e2e2e",
      "hover": {
        "bgColor": "#e8feff"
      }
    }
  },
  "connectionError": {
    "textColor": "#fff",
    "bgColor": "red"
  },
  "alerts": {
    "warning": {
      "color": "red"
    }
  },
  "PageHeader": {
    "bottomBorderColor": "#bbb"
  },
  "homePage": {
    "bgColor": "#eee",
    "navigation": {
      "bgColor": "#fff",
      "loginButton": {
        "default": {
          "loginButtonBgColor": "#fff",
          "loginButtonTextColor": "#1779ba",
          "loginButtonBorderColor": "#1779ba"
        },
        "hover": {
          "loginButtonBgColor": "#1779ba",
          "loginButtonTextColor": "#fff",
          "loginButtonBorderColor": "#1779ba"
        }
      }
    }
  },
  "dashboardPage": {
    "avatar": {
      "borderColor": "#1779ba"
    },
    "contextMenu": {
      "bgColor": "#fff",
      "borderColor": "#e5e5e5",
      "menuItemTextColor": "#2e2e2e",
      "menuItemBgColor": "#fff",
      "menuItemHoverBgColor": "#eee",
      "dividerBgColor": "#dfdfdf"
    }
  },
  "loginPage": {
    "bgColor": "#eee"
  },
  "userSettingsPage": {
    "bgColor": "#f1f1f1",
    "sidePanel": {
      "bgColor": "#e0e0e0",
      "sidePanelItem": {
        "bgColor": "#c7c7c7",
        "activeBgColor": "#f1f1f1",
        "highlightColor": "#1779ba",
        "fontColor": "#2E2E30"
      }
    }
  },
  "keyManagementPage": {
    "iconColor": "#1779ba",
    "passwordColor": "orange",
    "noActiveKeyWarning": {
      "textColor": "orange",
      "borderColor": "orange"
    }
  },
  "layoutPage": {
    "panel": {
      "bgColor": "#f1f1f1",
      "borderColor": "#e0e0e0",
      "contentSeparator": "#ccc"
    }
  }
}
