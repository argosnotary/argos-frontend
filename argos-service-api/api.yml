#
# Copyright (C) 2019 - 2020 Rabobank Nederland
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#         http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

openapi: "3.0.0"
info:
  version: 1.0.0
  title: Argos Service
  license:
    name: Apache License 2.0
servers:
  - url: /api
paths:
  /supplychain:
    post:
      summary: create a supplychain
      operationId: createSupplyChain
      tags:
        - supplychain
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/CreateSupplyChainCommand"
      responses:
        '201':
          description: created
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/SupplyChainItem"
          headers:
            Location:
              schema:
                type: string
                format: uri
              description: location where the created supplychain can be found
        '400':
          $ref: '#/components/responses/ValidationError'
        '500':
          $ref: '#/components/responses/Error'

    get:
      summary: search a supplychain
      operationId: searchSupplyChains
      tags:
        - supplychain
      parameters:
        - name: name
          in: query
          description: name of the supplychain
          required: false
          schema:
            type: string
      responses:
        '200':
          description: array of supply chain items
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/SupplyChainItem"
        '500':
          $ref: '#/components/responses/Error'

  /supplychain/{supplyChainId}:
    get:
      summary: get a supplychain item
      operationId: getSupplyChain
      tags:
        - supplychain
      parameters:
        - name: supplyChainId
          in: path
          description: supply chain id
          required: true
          schema:
            type: string
      responses:
        '200':
          description: the supply chain item
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/SupplyChainItem"
        '500':
          $ref: '#/components/responses/Error'

  /supplychain/{supplyChainId}/link:
    get:
      summary: find link
      operationId: findLink
      tags:
        - link
      parameters:
        - name: supplyChainId
          in: path
          description: supply chain id
          required: true
          schema:
            type: string
        - name: hash
          in: query
          description: hash of product or material
          required: false
          schema:
            type: string
      responses:
        '200':
          description: array of links
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/LinkMetaBlock"
        '500':
          $ref: '#/components/responses/Error'
    post:
      summary: Create a link
      operationId: createLink
      tags:
        - link
      parameters:
        - name: supplyChainId
          in: path
          description: supply chain id
          required: true
          schema:
            type: string
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/LinkMetaBlock"
      responses:
        '204':
          description: no content
        '400':
          $ref: '#/components/responses/ValidationError'
        '500':
          $ref: '#/components/responses/Error'
  /key:
    post:
      summary: stores a public and private key pair
      operationId: storeKey
      tags:
        - key
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/KeyPair"
      responses:
        '204':
          description: no content
        '400':
          $ref: '#/components/responses/ValidationError'
        '500':
          $ref: '#/components/responses/Error'
  /key/{keyId}:
    get:
      summary: get a key pair by id
      operationId: getKey
      tags:
        - key
      parameters:
        - name: keyId
          in: path
          description: key pair id
          required: true
          schema:
            type: string
            minLength: 64
            maxLength: 64
            pattern: '^[0-9a-f]*$'
      responses:
        '200':
          description: Key pair
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/KeyPair"
        '400':
          $ref: '#/components/responses/ValidationError'
        '500':
          $ref: '#/components/responses/Error'
  /supplychain/{supplyChainId}/layout:
    get:
      summary: get layout
      operationId: findLayout
      tags:
        - layout
      parameters:
        - name: supplyChainId
          in: path
          description: supply chain id
          required: true
          schema:
            type: string
      responses:
        '200':
          description: array of layouts
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: "#/components/schemas/LayoutMetaBlock"
        '500':
          $ref: '#/components/responses/Error'
    post:
      summary: Create a layout
      operationId: createLayout
      tags:
        - layout
      parameters:
        - name: supplyChainId
          in: path
          description: supply chain id
          required: true
          schema:
            type: string
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/LayoutMetaBlock"
      responses:
        '201':
          description: created
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/LayoutMetaBlock"
          headers:
            Location:
              schema:
                type: string
                format: uri
              description: location where the created layout metablock can be found
        '400':
          $ref: '#/components/responses/ValidationError'
        '500':
          $ref: '#/components/responses/Error'

  /supplychain/{supplyChainId}/layout/{layoutId}:
    get:
      summary: get layout
      operationId: getLayout
      tags:
        - layout
      parameters:
        - name: supplyChainId
          in: path
          description: supply chain id
          required: true
          schema:
            type: string
        - name: layoutId
          in: path
          description: layout id
          required: true
          schema:
            type: string
      responses:
        '200':
          description: layouts
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/LayoutMetaBlock"
        '500':
          $ref: '#/components/responses/Error'
    put:
      summary: update a layout
      operationId: updateLayout
      tags:
        - layout
      parameters:
        - name: supplyChainId
          in: path
          description: supply chain id
          required: true
          schema:
            type: string
        - name: layoutId
          in: path
          description: layout id
          required: true
          schema:
            type: string
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/LayoutMetaBlock"
      responses:
        '200':
          description: updated
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/LayoutMetaBlock"
        '400':
          $ref: '#/components/responses/ValidationError'
        '500':
          $ref: '#/components/responses/Error'

  /supplychain/{supplyChainId}/verification:
    post:
      summary: creates a verification run
      operationId: performVerification
      tags:
        - verification
      parameters:
        - name: supplyChainId
          in: path
          description: supply chain id
          required: true
          schema:
            type: string
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/VerifyCommand"
      responses:
        '200':
          description: VerificationResult
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/VerificationResult'
        '400':
          $ref: '#/components/responses/ValidationError'
        '500':
          $ref: '#/components/responses/Error'
components:
  responses:
    ValidationError:
      description: validation error
      content:
        application/json:
          schema:
            $ref: "#/components/schemas/Error"
    Error:
      description: server error
      content:
        application/json:
          schema:
            $ref: "#/components/schemas/Error"
  schemas:
    VerifyCommand:
      type: object
      required:
        - productsToVerify
      properties:
        expectedProducts:
          type: array
          items:
            $ref: "#/components/schemas/Artifact"

    VerificationResult:
      type: object
      required:
        - runIsValid
      properties:
        runIsValid:
          type: boolean

    LinkMetaBlock:
      type: object
      required:
        - signature
        - link
      properties:
        signature:
          $ref: "#/components/schemas/Signature"
        link:
          $ref: "#/components/schemas/Link"
    Signature:
      type: object
      required:
        - keyId
        - signature
      properties:
        keyId:
          $ref: '#/components/schemas/KeyId'
        signature:
          type: string
          minLength: 512
          maxLength: 1024
          pattern: '^[0-9a-f]*$'
    Link:
      type: object
      required:
        - runId
        - layoutSegmentName
        - stepName
        - command
        - materials
        - products
      properties:
        runId:
          type: string
        stepName:
          type: string
        layoutSegmentName:
          type: string
        command:
          type: array
          items:
            type: string
            pattern: '^[0-9a-f]*$'
        materials:
          type: array
          items:
            $ref: "#/components/schemas/Artifact"
        products:
          type: array
          items:
            $ref: "#/components/schemas/Artifact"
    Artifact:
      type: object
      required:
        - uri
        - hash
      properties:
        uri:
          type: string
          maxLength: 1024
          pattern: '^(?!.*\\).*$'
        hash:
          type: string
          minLength: 64
          maxLength: 64
          pattern: '^[0-9a-f]*$'
    KeyPair:
      type: object
      required:
        - keyId
        - publicKey
      properties:
        keyId:
          $ref: '#/components/schemas/KeyId'
        publicKey:
          type: string
          format: byte
        encryptedPrivateKey:
          type: string
          format: byte
    CreateSupplyChainCommand:
      type: object
      required:
        - name
      properties:
        name:
          type: string
          maxLength: 256
          description: name of the supply chain must be unique
    SupplyChainItem:
      type: object
      required:
        - id
        - name
      properties:
        id:
          type: string
          maxLength: 36
        name:
          type: string
          maxLength: 256
    LayoutMetaBlock:
      type: object
      required:
        - signatures
        - layout
      properties:
        id:
          type: string
        signatures:
          type: array
          items:
            $ref: "#/components/schemas/Signature"
        layout:
          $ref: "#/components/schemas/Layout"
    Layout:
      type: object
      required:
        - authorizedKeyIds
        - expectedEndProducts
        - layoutSegments
      properties:
        authorizedKeyIds:
          type: array
          minItems: 1
          items:
            $ref: '#/components/schemas/KeyId'
        expectedEndProducts:
          type: array
          minItems: 1
          items:
            $ref: '#/components/schemas/MatchRule'
        layoutSegments:
          type: array
          minItems: 1
          items:
            $ref: "#/components/schemas/LayoutSegment"
    LayoutSegment:
      type: object
      required:
        - name
        - steps
      properties:
        name:
          type: string
        steps:
          type: array
          minItems: 1
          items:
            $ref: "#/components/schemas/Step"
    MatchRule:
      type: object
      required:
        - pattern
        - destinationType
        - destinationSegmentName
        - destinationStepName
      properties:
        pattern:
          type: string
        destinationType:
          type: string
          enum:
            - PRODUCTS
            - MATERIALS
        destinationSegmentName:
          type: string
        destinationStepName:
          type: string
    Step:
      type: object
      required:
        - name
        - requiredSignatures
        - authorizedKeyIds
        - expectedMaterials
        - expectedProducts
        - expectedCommand
      properties:
        name:
          type: string
        authorizedKeyIds:
          type: array
          minItems: 1
          items:
            $ref: '#/components/schemas/KeyId'
        requiredNumberOfLinks:
          type: integer
          format: int32
          minimum: 1
        expectedCommand:
          type: array
          items:
            type: string
            pattern: '^[0-9a-f]*$'
        expectedMaterials:
          type: array
          items:
            $ref: "#/components/schemas/Rule"
        expectedProducts:
          type: array
          items:
            $ref: "#/components/schemas/Rule"
    KeyId:
      type: string
      description: key id is a hash of the public key
      minLength: 64
      maxLength: 64
      pattern: '^[0-9a-f]*$'
    Rule:
      type: object
      required:
        - ruleType
        - pattern
      properties:
        ruleType:
          type: string
          enum:
            - ALLOW
            - CREATE
            - DELETE
            - DISALLOW
            - MATCH
            - MODIFY
            - REQUIRE
        pattern:
          type: string
        sourcePathPrefix:
          type: string
        destinationPathPrefix:
          type: string
        destinationType:
          type: string
          enum:
            - PRODUCTS
            - MATERIALS
        destinationStepName:
          type: string
    Error:
      type: object
      required:
        - message
      properties:
        message:
          type: string
