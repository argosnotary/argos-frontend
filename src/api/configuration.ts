/*
 * Argos Notary - A new way to secure the Software Supply Chain
 *
 * Copyright (C) 2019 - 2020 Rabobank Nederland
 * Copyright (C) 2019 - 2021 Gerard Borst <gerard.borst@argosnotary.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
/* tslint:disable */
/* eslint-disable */
/**
 * Argos Service
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


export interface ConfigurationParameters {
    apiKey?: string | Promise<string> | ((name: string) => string) | ((name: string) => Promise<string>);
    username?: string;
    password?: string;
    accessToken?: string | Promise<string> | ((name?: string, scopes?: string[]) => string) | ((name?: string, scopes?: string[]) => Promise<string>);
    basePath?: string;
    baseOptions?: any;
    formDataCtor?: new () => any;
}

export class Configuration {
    /**
     * parameter for apiKey security
     * @param name security name
     * @memberof Configuration
     */
    apiKey?: string | Promise<string> | ((name: string) => string) | ((name: string) => Promise<string>);
    /**
     * parameter for basic security
     *
     * @type {string}
     * @memberof Configuration
     */
    username?: string;
    /**
     * parameter for basic security
     *
     * @type {string}
     * @memberof Configuration
     */
    password?: string;
    /**
     * parameter for oauth2 security
     * @param name security name
     * @param scopes oauth2 scope
     * @memberof Configuration
     */
    accessToken?: string | Promise<string> | ((name?: string, scopes?: string[]) => string) | ((name?: string, scopes?: string[]) => Promise<string>);
    /**
     * override base path
     *
     * @type {string}
     * @memberof Configuration
     */
    basePath?: string;
    /**
     * base options for axios calls
     *
     * @type {any}
     * @memberof Configuration
     */
    baseOptions?: any;
    /**
     * The FormData constructor that will be used to create multipart form data
     * requests. You can inject this here so that execution environments that
     * do not support the FormData class can still run the generated client.
     *
     * @type {new () => FormData}
     */
    formDataCtor?: new () => any;

    constructor(param: ConfigurationParameters = {}) {
        this.apiKey = param.apiKey;
        this.username = param.username;
        this.password = param.password;
        this.accessToken = param.accessToken;
        this.basePath = param.basePath;
        this.baseOptions = param.baseOptions;
        this.formDataCtor = param.formDataCtor;
    }

    /**
     * Check if the given MIME is a JSON MIME.
     * JSON MIME examples:
     *   application/json
     *   application/json; charset=UTF8
     *   APPLICATION/JSON
     *   application/vnd.company+json
     * @param mime - MIME (Multipurpose Internet Mail Extensions)
     * @return True if the given MIME is JSON, false otherwise.
     */
    public isJsonMime(mime: string): boolean {
        const jsonMime: RegExp = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
        return mime !== null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
    }
}
