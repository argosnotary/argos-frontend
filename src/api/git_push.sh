#!/bin/sh
#
# Argos Notary - A new way to secure the Software Supply Chain
#
# Copyright (C) 2019 - 2020 Rabobank Nederland
# Copyright (C) 2019 - 2021 Gerard Borst <gerard.borst@argosnotary.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.
#

# ref: https://help.github.com/articles/adding-an-existing-project-to-github-using-the-command-line/
#
# Usage example: /bin/sh ./git_push.sh wing328 openapi-pestore-perl "minor update" "gitlab.com"

git_user_id=$1
git_repo_id=$2
release_note=$3
git_host=$4

if [ "$git_host" = "" ]; then
    git_host="github.com"
    echo "[INFO] No command line input provided. Set \$git_host to $git_host"
fi

if [ "$git_user_id" = "" ]; then
    git_user_id="GIT_USER_ID"
    echo "[INFO] No command line input provided. Set \$git_user_id to $git_user_id"
fi

if [ "$git_repo_id" = "" ]; then
    git_repo_id="GIT_REPO_ID"
    echo "[INFO] No command line input provided. Set \$git_repo_id to $git_repo_id"
fi

if [ "$release_note" = "" ]; then
    release_note="Minor update"
    echo "[INFO] No command line input provided. Set \$release_note to $release_note"
fi

# Initialize the local directory as a Git repository
git init

# Adds the files in the local repository and stages them for commit.
git add .

# Commits the tracked changes and prepares them to be pushed to a remote repository.
git commit -m "$release_note"

# Sets the new remote
git_remote=`git remote`
if [ "$git_remote" = "" ]; then # git remote not defined

    if [ "$GIT_TOKEN" = "" ]; then
        echo "[INFO] \$GIT_TOKEN (environment variable) is not set. Using the git credential in your environment."
        git remote add origin https://${git_host}/${git_user_id}/${git_repo_id}.git
    else
        git remote add origin https://${git_user_id}:${GIT_TOKEN}@${git_host}/${git_user_id}/${git_repo_id}.git
    fi

fi

git pull origin master

# Pushes (Forces) the changes in the local repository up to the remote repository
echo "Git pushing to https://${git_host}/${git_user_id}/${git_repo_id}.git"
git push origin master 2>&1 | grep -v 'To https'

