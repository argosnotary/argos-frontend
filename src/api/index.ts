/*
 * Argos Notary - A new way to secure the Software Supply Chain
 *
 * Copyright (C) 2019 - 2020 Rabobank Nederland
 * Copyright (C) 2019 - 2021 Gerard Borst <gerard.borst@argosnotary.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
/* tslint:disable */
/* eslint-disable */
/**
 * Argos Service
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


export * from "./api";
export * from "./configuration";

