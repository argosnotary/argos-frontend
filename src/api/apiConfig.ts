import { Configuration } from "./configuration";

export function getApiConfigUnauthenticated() {
  return new Configuration({ basePath: "/api" });
}

export function getApiConfig(token: string) {
  return new Configuration({ basePath: "/api", accessToken: token });
}
