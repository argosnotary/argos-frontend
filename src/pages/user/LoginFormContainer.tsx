/*
 * Copyright (C) 2020 Argos Notary
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React, { useEffect, useState } from "react";

import LoginForm from "./LoginForm";
import axios from "axios";
import { OauthProviderApi, OAuthProvider } from "../../api/api";
import { getApiConfigUnauthenticated } from "../../api/apiConfig";

function LoginFormContainer(): JSX.Element {
  const [providers, setProviders] = useState<Array<OAuthProvider>>([]);
  const [showError, setShowError] = useState(false);

  useEffect(() => {
    const source = axios.CancelToken.source();
    const api = new OauthProviderApi(getApiConfigUnauthenticated());

    api
      .getOAuthProviders()
      .then(response => {
        setProviders(response.data);
      })
      .catch(error => {
        if (!axios.isCancel(error)) {
          setShowError(true);
        }
      });

    return () => {
      source.cancel();
    };
  }, []);

  return <LoginForm showError={showError} providers={providers} />;
}

export default LoginFormContainer;
