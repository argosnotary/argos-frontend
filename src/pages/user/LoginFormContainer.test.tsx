/*
 * Copyright (C) 2020 Argos Notary
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import React from "react";
import { mount } from "enzyme";

import MockAdapter from "axios-mock-adapter";
import Axios from "axios";
import { ThemeProvider } from "styled-components";
import theme from "../../theme/base.json";
import { act } from "react-dom/test-utils";
import LoginFormContainer from "./LoginFormContainer";
import { OAuthProvider } from "../../api/api";

const mock = new MockAdapter(Axios);

jest.mock("react-router-dom", () => {
    return {
        useHistory: jest.fn()
    };
});

it("renders correctly", async () => {
    const providers: Array<OAuthProvider> = [
        {
            providerName: "provider1",
            displayName: "name1"
        },
        {
            providerName: "provider2",
            displayName: "name2",
            iconUrl: "http://localhost/icon.svg"
        }
    ];

    mock.onGet("/api/oauthprovider").reply(200, providers);

    const root = mount(
        <ThemeProvider theme={theme}>
            <LoginFormContainer />
        </ThemeProvider>
    );

    await act(() =>
        new Promise(resolve => setImmediate(resolve)).then(() => {
            root.update();
            expect(root.find(LoginFormContainer)).toMatchSnapshot();
        })
    );
});

it("renders correctly error", async () => {
    mock.onGet("/api/oauthprovider").reply(404);

    const root = mount(
        <ThemeProvider theme={theme}>
            <LoginFormContainer />
        </ThemeProvider>
    );

    await act(() =>
        new Promise(resolve => setImmediate(resolve)).then(() => {
            root.update();
            expect(root.find(LoginFormContainer)).toMatchSnapshot();
        })
    );
});
