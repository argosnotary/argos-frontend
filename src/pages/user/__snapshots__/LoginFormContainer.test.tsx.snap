// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<LoginFormContainer>
  <LoginFormContainer
    providers={
      Array [
        Object {
          "displayName": "name1",
          "providerName": "provider1",
        },
        Object {
          "displayName": "name2",
          "iconUrl": "http://localhost/icon.svg",
          "providerName": "provider2",
        },
      ]
    }
    showError={false}
  >
    <styled.div>
      <div
        className="sc-jrAGrp hhLMWH"
      >
        <styled.img
          src="images/logo.svg"
        >
          <img
            className="sc-pFZIQ kfQfzL"
            src="images/logo.svg"
          />
        </styled.img>
        <Styled(styled.a)
          href="/api/oauth2/authorize/provider1?redirect_uri=/authenticated"
          key="provider0"
        >
          <a
            className="sc-dlfnbm sc-fubCfw fUfJKR jJDjXe"
            href="/api/oauth2/authorize/provider1?redirect_uri=/authenticated"
          >
            <styled.span>
              <span
                className="sc-gKsewC"
              >
                Login with 
                name1
              </span>
            </styled.span>
          </a>
        </Styled(styled.a)>
        <Styled(styled.a)
          href="/api/oauth2/authorize/provider2?redirect_uri=/authenticated"
          key="provider1"
        >
          <a
            className="sc-dlfnbm sc-fubCfw fUfJKR jJDjXe"
            href="/api/oauth2/authorize/provider2?redirect_uri=/authenticated"
          >
            <styled.img
              src="http://localhost/icon.svg"
            >
              <img
                className="sc-jSgupP iXgOms"
                src="http://localhost/icon.svg"
              />
            </styled.img>
            <styled.span>
              <span
                className="sc-iBPRYJ gbApWE"
              />
            </styled.span>
            <styled.span>
              <span
                className="sc-gKsewC"
              >
                Login with 
                name2
              </span>
            </styled.span>
          </a>
        </Styled(styled.a)>
      </div>
    </styled.div>
  </LoginFormContainer>
</LoginFormContainer>
`;

exports[`renders correctly error 1`] = `
<LoginFormContainer>
  <LoginFormContainer
    providers={Array []}
    showError={true}
  >
    <styled.div>
      <div
        className="sc-jrAGrp hhLMWH"
      >
        <styled.img
          src="images/logo.svg"
        >
          <img
            className="sc-pFZIQ kfQfzL"
            src="images/logo.svg"
          />
        </styled.img>
        <styled.p>
          <p
            className="sc-bdfBwQ cQaTwN"
          >
            Connection error try again later
          </p>
        </styled.p>
      </div>
    </styled.div>
  </LoginFormContainer>
</LoginFormContainer>
`;
